### PR TITLE
Remove superflous parameter `overseer_enable_anyways` and make parachain node type more explicit

### PR DESCRIPTION
**Pull request by bkchr**
This pull request removes the superfluous parameter  `overseer_enable_anyways` and renames `IsCollator` to `IsParachainNode`. The parameter `overseer_enable_anyways` was actually used to express that a relay chain node is running alongside a parachain full node. This lead to things like enabling of the candidate validation system running, which is only required for relay chain validators. This pull requests makes this more explicit by introducing `IsParachainNode::FullNode` and ensuring that we don't enable subsystems that we don't need for parachain nodes.

Cumulus companion: https://github.com/paritytech/cumulus/pull/3008

### CODE REVIEW

**Comment by skunert** [comment]
What about pov-recovery for parachain full nodes? AFAIR that was the reason for this flag.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3383072

**Comment by bkchr** [comment]
> What about pov-recovery for parachain full nodes? AFAIR that was the reason for this flag.

Ahh yes, good point! Reworked the pr.

**Comment by pepoviola** [comment]
@bkchr zombienet tests are failing with this error, looks like an issue in the collator image.

```
Error: Input("Worker binaries could not be found, make sure polkadot was built/installed correctly. If you ran with `cargo run`, please run `cargo build` first. Searched given workers path (None), polkadot binary path (\"/usr/local/bin\"), and lib path (/usr/lib/polkadot), workers names: None")
```
Logs:
https://grafana.teleport.parity.io/goto/SXuFYj64g?orgId=1
https://grafana.teleport.parity.io/goto/JnZaLjeVg?orgId=1

I will take a look. Thx!

**Comment by skunert** [comment]
> https://grafana.teleport.parity.io/goto/SXuFYj64g?orgId=1

I think this is because the undying collator is started with `--validator` which makes it an authority. Together with this change https://github.com/paritytech/polkadot/pull/7617/files#diff-6dcbafad4c91afb2c91f5b01429549242a45a3bb0de721c830abd0a61421ae3aL913 it will try to start the workers.

**Review by skunert — APPROVED** [review]

**Comment by bkchr** [comment]
@pepoviola fixed the CI failure. It is related to how zombienet spawns the node, as @skunert explains above. I "fixed" this for now. The proper solution would be that zombienet is not using "default cli args", but that is something for another day :P

**Comment by skunert** [comment]
> @pepoviola fixed the CI failure. It is related to how zombienet spawns the node, as @skunert explains above. I "fixed" this for now. The proper solution would be that zombienet is not using "default cli args", but that is something for another day :P

Still slightly confused by this, `validator` should not be default afaik. Also, the behaviour here is a bit different because in cumulus we handle `--validator` as `--collator`. Just wondering if setting `cumulus_based=false` would help here. Just curious why zombienet is passing `--validator` in the first place 🤔

**Comment by bkchr** [comment]
bot merge

**Comment by pepoviola** [comment]
Thanks @bkchr / @skunert / @mrcnski, I created this [followup issue](https://github.com/paritytech/zombienet/issues/1245) to fix this in zombienet.

**Comment by rphmeier** [comment]
This shouldn't have been merged IMO

Subsystems shouldn't be disabling themselves internally! Just don't add the subsystem if you don't need it.

And we're baking in all of these half-formed assumptions about what a parachain node is or what a collator is. These assumptions will slow us down later. The service instantiation is already a petrified lump. If it could speak, it'd say "kill me."

File: parachain/test-parachains/adder/collator/src/main.rs [hunk 1]
```diff
@@ -57,10 +58,15 @@ fn main() -> Result<()> {
 				let collator = Collator::new();
 
 				config.disable_beefy = true;
+				// Zombienet is spawning all collators currently with the same CLI, this means it
+				// sets `--validator` and this is wrong here.
+				config.role = Role::Full;
```
[the remark this answers is not in the document]
Comment (pepoviola) [reply parachain/test-parachains/adder/collator/src/main.rs:63]: Fixed [here](https://github.com/paritytech/zombienet/pull/1247), I will create a new `pr` to remove this workaroud.

**Comment by rphmeier** [comment]
ref #7576

**Pull request merged**